server/util/flagutil/yaml/yaml.go

```diff
@@ -648,11 +728,8 @@ func RetypeAndFilterYAMLMap(yamlMap map[string]any, typeMap map[string]any, pref
 
 // OverrideFlagsFromData takes some YAML input and marshals it, then uses the
 // unmarshaled data to override the flags with names corresponding to the keys.
-func OverrideFlagsFromData(data []byte) error {
-	// expand environment variables
-	expandedData := []byte(os.ExpandEnv(string(data)))
-
-	yamlMap, node, err := getYAMLMapAndNodeFromData(expandedData)
+func OverrideFlagsFromData(data string) error {
```

Comment:
Since the expansion was moved to after the call to `yaml.Unmarshal`, there's no longer a compelling reason to change this to `string` from `[]byte`, since it is never used anywhere as a string, and it is used as `[]byte`. Can we switch it back?

Reply:
Having this be a string does make it nicer to use in tests since there's no need to convert the test config from []byte to string.

Reply:
Fair enough; it's cosmetic either way, I suppose.